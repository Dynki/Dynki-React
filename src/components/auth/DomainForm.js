import React from 'react';
import { Form, Icon, Input, Button } from 'antd';

const FormItem = Form.Item;

export const DForm = Form.create({
    onValuesChange(props, values) {
        props.onChange(values.name);
    }
})((props) => {
    const { getFieldDecorator } = props.form;

    const handleSubmit = (e) => {
        e.preventDefault();
        props.form.validateFields((err, values) => {
            if (!err) {
                console.log('Received values of form: ', values);
                props.onCreateDomain(values.name)
            }
        });
    }

    const validateTeamName = (rule, value, callback) => {

        const isValid = value => {
            return new RegExp(/^(\d|\w)+$/).test(value);
        }
 
        if (isValid(value)) {
            callback();
        } else {
            callback('No spaces or special characters');
        }

    } 

    return (
        <Form className="new_domain__form" onSubmit={handleSubmit}>
            <FormItem 
                hasFeedback
                validateStatus={props.validateStatus}
                help={props.validateFeedback}
            >
                {getFieldDecorator('name', {
                    rules: [
                        { required: true, message: "We're gonna need a team name" },
                        { validator: validateTeamName },
                        { max: 50, message: 'Toooo loooonnnnggg!!' },
                        { min: 3, message: 'Not long enough, try a longer name!' }
                    ],
                })(
                    <Input 
                        size="large"
                        prefix={<Icon type="team"
                        style={{ color: 'rgba(0,0,0,.25)' }} />}
                        placeholder="Enter your team name"
                        autoComplete="off" 
                    />
                )}
            </FormItem>
            <FormItem>
                <Button id="btnCreateTeam" type="dashed" htmlType="submit" className="domain__btn" loading={props.pending} disabled={props.pending}>
                    New Team
                    <Icon type="arrow-right" />
                </Button>
            </FormItem>
        </Form>
    )
});

const DomainForm = (props) => {

    const handleFormChange = (name) => {
        // props.onCheckDomain(name);
    }

    return (
        <div style={{width: "100%" }}>
            <DForm {...props} onChange={handleFormChange} />
        </div>
    );
}


export default DomainForm;
