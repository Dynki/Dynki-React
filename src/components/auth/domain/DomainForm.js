import React from 'react';
import { Form, Icon, Input, Button } from 'antd';

const FormItem = Form.Item;

const DomainForm = ({ form, onUpdateDomain, pending, validateFeedback, validateStatus }) => {
    const { getFieldDecorator } = form;

    const handleSubmit = (e) => {
        e.preventDefault();
        form.validateFields((err, values) => {
            if (!err) {
                onUpdateDomain(values.name)
            }
        });
    }

    return (
        <Form className="new_domain__form" onSubmit={handleSubmit}>
            <FormItem 
                hasFeedback
                validateStatus={validateStatus}
                help={validateFeedback}
            >
                {getFieldDecorator('name', {
                    rules: [
                        { required: true, message: "We're gonna need a team name" },
                        { max: 50, message: 'Toooo loooonnnnggg!!' },
                        { min: 3, message: 'Not long enough, try a longer name!' }
                    ],
                })(
                    <Input 
                        size="large"
                        autoFocus
                        disabled={pending}
                        prefix={<Icon type="team"
                        style={{ color: 'rgba(0,0,0,.25)' }} />}
                        placeholder="Enter your team name"
                        autoComplete="off" 
                    />
                )}
            </FormItem>
            <FormItem>
                <Button id="btnCreateTeam" type="dashed" htmlType="submit" className="button" loading={pending} disabled={pending}>
                    Create Team
                </Button>
            </FormItem>
        </Form>
    )
};

export default Form.create()(DomainForm);
