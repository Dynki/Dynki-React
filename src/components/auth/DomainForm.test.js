import React from 'react';
import { shallow, mount } from 'enzyme';
import DomainForm, { DForm } from './DomainForm';
import { Form } from 'antd';

/**
 * DomainForm
 */

it('should render full component when domain has been checked', () => {
    const wrapper = shallow(<DomainForm />);
    expect(wrapper).toMatchSnapshot();
});

it('should call crate domain with correct name', () => {
    const mockCheckDomain = jest.fn();
    const props = {
        onCheckDomain: mockCheckDomain
    }
    const wrapper = shallow(<DomainForm {...props} />);
    wrapper.props().children.props.onChange();

    expect(mockCheckDomain).toHaveBeenCalledTimes(1);
});

/**
 * DForm
 */
it('should render full component when domain has been checked', () => {
    let formRef;
    const EnhancedForm = Form.create()(DForm);
    const wrapper = mount(
        <EnhancedForm wrappedComponentRef={form => (formRef = form)} />
    );

    expect(wrapper).toMatchSnapshot();
});

it('should handle form submission when fields are valid', () => {
    const mockCreateDomain = jest.fn();
    let formRef;
    const EnhancedForm = Form.create()(DForm);
    const wrapper = mount(
        <EnhancedForm onCreateDomain={mockCreateDomain} wrappedComponentRef={form => (formRef = form)} />
    );

    const event = {
        preventDefault: jest.fn(),
        target: {}
    };

    const passwordInput = wrapper.find('input');
    passwordInput.instance().value = 'TestTeam';
    passwordInput.simulate('change');
    wrapper.find('form').simulate('submit', event);

    expect(mockCreateDomain).toHaveBeenCalledTimes(1);
});

it('should NOT handle form submission when fields are invalid', () => {
    const mockCreateDomain = jest.fn();
    let formRef;
    const EnhancedForm = Form.create()(DForm);
    const wrapper = mount(
        <EnhancedForm onCreateDomain={mockCreateDomain} wrappedComponentRef={form => (formRef = form)} />
    );

    const event = {
        preventDefault: jest.fn(),
        target: {}
    };

    wrapper.find('form').simulate('submit', event);

    expect(mockCreateDomain).toHaveBeenCalledTimes(0);
});
