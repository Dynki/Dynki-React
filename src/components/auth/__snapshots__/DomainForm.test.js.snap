// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render full component when domain has been checked 1`] = `
<div
  style={
    Object {
      "width": "100%",
    }
  }
>
  <Form(WrappedComponent)
    onChange={[Function]}
  />
</div>
`;

exports[`should render full component when domain has been checked 2`] = `
<form
  className="ant-form ant-form-horizontal new_domain__form"
  onSubmit={[Function]}
>
  <div
    className="ant-row ant-form-item"
  >
    <div
      className="ant-form-item-control-wrapper"
    >
      <div
        className="ant-form-item-control"
      >
        <span
          className="ant-form-item-children"
        >
          <span
            className="ant-input-affix-wrapper ant-input-affix-wrapper-lg"
          >
            <span
              className="ant-input-prefix"
            >
              <i
                aria-label="icon: team"
                className="anticon anticon-team"
                style={
                  Object {
                    "color": "rgba(0,0,0,.25)",
                  }
                }
              >
                <svg
                  aria-hidden="true"
                  className=""
                  data-icon="team"
                  fill="currentColor"
                  height="1em"
                  key="svg-team"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M824.2 699.9a301.55 301.55 0 0 0-86.4-60.4C783.1 602.8 812 546.8 812 484c0-110.8-92.4-201.7-203.2-200-109.1 1.7-197 90.6-197 200 0 62.8 29 118.8 74.2 155.5a300.95 300.95 0 0 0-86.4 60.4C345 754.6 314 826.8 312 903.8a8 8 0 0 0 8 8.2h56c4.3 0 7.9-3.4 8-7.7 1.9-58 25.4-112.3 66.7-153.5A226.62 226.62 0 0 1 612 684c60.9 0 118.2 23.7 161.3 66.8C814.5 792 838 846.3 840 904.3c.1 4.3 3.7 7.7 8 7.7h56a8 8 0 0 0 8-8.2c-2-77-33-149.2-87.8-203.9zM612 612c-34.2 0-66.4-13.3-90.5-37.5a126.86 126.86 0 0 1-37.5-91.8c.3-32.8 13.4-64.5 36.3-88 24-24.6 56.1-38.3 90.4-38.7 33.9-.3 66.8 12.9 91 36.6 24.8 24.3 38.4 56.8 38.4 91.4 0 34.2-13.3 66.3-37.5 90.5A127.3 127.3 0 0 1 612 612zM361.5 510.4c-.9-8.7-1.4-17.5-1.4-26.4 0-15.9 1.5-31.4 4.3-46.5.7-3.6-1.2-7.3-4.5-8.8-13.6-6.1-26.1-14.5-36.9-25.1a127.54 127.54 0 0 1-38.7-95.4c.9-32.1 13.8-62.6 36.3-85.6 24.7-25.3 57.9-39.1 93.2-38.7 31.9.3 62.7 12.6 86 34.4 7.9 7.4 14.7 15.6 20.4 24.4 2 3.1 5.9 4.4 9.3 3.2 17.6-6.1 36.2-10.4 55.3-12.4 5.6-.6 8.8-6.6 6.3-11.6-32.5-64.3-98.9-108.7-175.7-109.9-110.9-1.7-203.3 89.2-203.3 199.9 0 62.8 28.9 118.8 74.2 155.5-31.8 14.7-61.1 35-86.5 60.4-54.8 54.7-85.8 126.9-87.8 204a8 8 0 0 0 8 8.2h56.1c4.3 0 7.9-3.4 8-7.7 1.9-58 25.4-112.3 66.7-153.5 29.4-29.4 65.4-49.8 104.7-59.7 3.9-1 6.5-4.7 6-8.7z"
                    key="svg-team-svg-0"
                  />
                </svg>
              </i>
            </span>
            <input
              autoComplete="off"
              className="ant-input ant-input-lg"
              data-__field={
                Object {
                  "name": "name",
                }
              }
              data-__meta={
                Object {
                  "name": "name",
                  "originalProps": Object {
                    "autoComplete": "off",
                    "disabled": false,
                    "placeholder": "Enter your team name",
                    "prefix": <Icon
                      style={
                        Object {
                          "color": "rgba(0,0,0,.25)",
                        }
                      }
                      type="team"
                    />,
                    "size": "large",
                    "type": "text",
                  },
                  "ref": null,
                  "rules": Array [
                    Object {
                      "message": "We're gonna need a team name",
                      "required": true,
                    },
                  ],
                  "trigger": "onChange",
                  "validate": Array [
                    Object {
                      "rules": Array [
                        Object {
                          "message": "We're gonna need a team name",
                          "required": true,
                        },
                      ],
                      "trigger": Array [
                        "onChange",
                      ],
                    },
                  ],
                  "valuePropName": "value",
                }
              }
              disabled={false}
              id="name"
              onChange={[Function]}
              onKeyDown={[Function]}
              placeholder="Enter your team name"
              size="large"
              style={null}
              type="text"
              value=""
            />
          </span>
        </span>
      </div>
    </div>
  </div>
  <button
    className="ant-btn domain__btn ant-btn-dashed"
    disabled={true}
    onClick={[Function]}
    type="submit"
  >
    <span
      key=".0"
    >
      Create Team
    </span>
    <i
      aria-label="icon: arrow-right"
      className="anticon anticon-arrow-right"
    >
      <svg
        aria-hidden="true"
        className=""
        data-icon="arrow-right"
        fill="currentColor"
        height="1em"
        key="svg-arrow-right"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M869 487.8L491.2 159.9c-2.9-2.5-6.6-3.9-10.5-3.9h-88.5c-7.4 0-10.8 9.2-5.2 14l350.2 304H152c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h585.1L386.9 854c-5.6 4.9-2.2 14 5.2 14h91.5c1.9 0 3.8-.7 5.2-2L869 536.2a32.07 32.07 0 0 0 0-48.4z"
          key="svg-arrow-right-svg-0"
        />
      </svg>
    </i>
  </button>
</form>
`;
